### PR TITLE
Add JetBrains-Toolbox extension

### PR DESCRIPTION
**Pull request by scomans**
Extension for JetBrains Toolbox that shows recent projects based on the [VSCode Extension](https://github.com/oliverschwendener/ueli/pull/1157) from [IrishBruse](https://github.com/IrishBruse)

Not tested on MacOS and Linux.

### CODE REVIEW

File: src/main/Extensions/JetBrainsToolbox/JetBrainsToolboxExtension.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+            defaultAction: {
+                handlerId: "Commandline",
+                description: t("openWith", { project: recent.name, toolName: recent.toolName }),
+                argument: `${recent.toolId.toLowerCase()} ${path}`,
+            },
+        };
+    }
+
+    public isSupported(): boolean {
+        return ["macOS", "Linux", "Windows"].includes(this.operatingSystem);
```
Comment (oliverschwendener) [inline src/main/Extensions/JetBrainsToolbox/JetBrainsToolboxExtension.ts:106]: Here you can just return `true`.

Comment (scomans) [reply src/main/Extensions/JetBrainsToolbox/JetBrainsToolboxExtension.ts]: Done

File: src/main/Extensions/JetBrainsToolbox/JetBrainsToolboxExtension.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    private async getRecents(): Promise<JetBrainsToolboxRecent[]> {
+        const cachePath = this.cachePaths[this.operatingSystem];
+        const projectsPath = join(cachePath, "intellij_projects.json");
+
+        // check if file exists
+        try {
+            await stat(projectsPath);
+        } catch (e) {
+            return [];
+        }
```
Comment (oliverschwendener) [inline src/main/Extensions/JetBrainsToolbox/JetBrainsToolboxExtension.ts:79]: You can use the `FileSystemUtility` to check if a file or folder exists:

```typescript
import type { FileSystemUtility } from "@Core/FileSystemUtility";

export class MyClass {
    public constructor(private readonly fileSystemUtility: FileSystemUtility) {}

    private async getRecents(): Promise<Recent[]> {
        const filePath = '...';

        const fileExists = await this.fileSystemUtility.pathExists(filePath);

        if (!fileExists) {
            return [];
        }

        // continue
    }
}
```

File: src/main/Extensions/JetBrainsToolbox/JetBrainsToolboxExtension.ts [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+interface JetBrainsToolboxRecent {
+    name: string;
+    path: string;
+    toolId: string;
+    toolName: string;
+    hasIcon: boolean;
+}
+
+export class JetBrainsToolboxExtension implements Extension {
+    public readonly id = "JetBrains-Toolbox";
```
[the remark this answers is not in the document]
Comment (scomans) [reply src/main/Extensions/JetBrainsToolbox/JetBrainsToolboxExtension.ts]: Sure :)

**Review by oliverschwendener — APPROVED** [review]

**Comment by oliverschwendener** [comment]
Thanks for your contribution! Code looks good, i'll quickly check if it works and then merge it.

**Comment by scomans** [comment]
I guessed the paths for macOS and Linux based on the paths for windows. Thanks for testing it on macOS. I installed it on a linux machine so that path should also be correct now.

**Review by oliverschwendener — APPROVED** [review]
Nice, it works now as expected 💪

**Comment by oliverschwendener** [comment]
I invited you to this repo and added you as code owner of this extension, so you get notifications when someone wants to change something in your extension.

**Pull request merged**